package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,9 +180,51 @@ def parse(self, **kwargs):
         resids = []
         resnames = []
         charges = []
+        has_charges = True
+        if sections['molecule'][3].strip() == 'NO_CHARGES':
+            has_charges = False
 
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
+            columns = a.split()
+            if len(columns) >= 9:
+                aid, name, x, y, z, atom_type, \
+                    resid, resname, charge = columns[:9]
+            elif len(columns) < 6:
+                raise ValueError(f"The @<TRIPOS>ATOM block in mol2 file"
+                                 f" {os.path.basename(self.filename)}"
+                                 f" should have at least 6 fields to be"
+                                 f" unpacked: atom_id atom_name x y z"
+                                 f" atom_type [subst_id[subst_name"
+                                 f" [charge [status_bit]]]]")
+            else:
+                aid, name, x, y, z, atom_type = columns[:6]
+                resid = 1
+                resname = None
+                charge = None
+                opt_values = [resid, resname, charge]
+                for i in range(6, len(columns)):
+                    opt_values[i-6] = columns[i]
+                resid, resname, charge = opt_values
+            # if has_charges:
+            #     if charge is None:
+            #         raise ValueError(f"The mol2 file {self.filename}"
+            #                          f" indicates no charges, but charge"
+            #                          f" provided in line: {a}.")
+            # else:
+            #     if charge is not None:
+            #         raise ValueError(f"The mol2 file {self.filename}"
+            #                          f" indicates a charge model"
+            #                          f"{sections['molecule'][3]}, but"
+            #                          f" no charge provided in line: {a}")
+            if charge is not None and (not has_charges):
+                raise ValueError(f"The mol2 file {self.filename}"
+                                 f" indicates no charges, but charge"
```

Comment:
```suggestion
                                 f" indicates no charges, but charge {charge}"
```
Maybe worth printing out the value of the charge as well?